#*Case Study Data Products*

#A. Fitur Extract

In [138]:
import pandas as pd
import pytz
from datetime import datetime
import re #untuk regex findall
import glob

In [139]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [140]:
#buat function untuk meng extract semua file dalam direktori tertentu
def exctractDataCsv(path):
    # mendapatkan list dari file di direktori
    listFiles = glob.glob(path)

    # membuat dataframe kosongan sbg tempat data yg digabung dimasukkan
    concatedData1 = pd.DataFrame()

    # looping setiap file di direktori tsb dan isinya di append (ditambahkan) di dataframe gabunga hasil concat
    for i in listFiles:
        data1 = pd.read_csv(i)
        concatedData1 = pd.concat([concatedData1, data1])

    #index tidak ditampilkan
    blanked_index=[''] * len(concatedData1)
    concatedData1.index=blanked_index

    #return data yang sudah di concate dan disembunyikan indexnya
    return concatedData1


#panggil function untuk extractDataCSV untuk semua file csv yang ada di direktori data_products
concatedData1 = exctractDataCsv('/content/drive/My Drive/purwadhika_files/data_sources/data_products/*.csv')
concatedData1.info()




<class 'pandas.core.frame.DataFrame'>
Index: 1103170 entries,  to 
Data columns (total 10 columns):
 #   Column          Non-Null Count    Dtype  
---  ------          --------------    -----  
 0   name            1103170 non-null  object 
 1   main_category   1103170 non-null  object 
 2   sub_category    1103170 non-null  object 
 3   image           1103170 non-null  object 
 4   link            1103170 non-null  object 
 5   ratings         751582 non-null   object 
 6   no_of_ratings   751582 non-null   object 
 7   discount_price  980844 non-null   object 
 8   actual_price    1067544 non-null  object 
 9   Unnamed: 0      551585 non-null   float64
dtypes: float64(1), object(9)
memory usage: 92.6+ MB


In [141]:
concatedData1.head()

name    main_category  \
  SOFTSPUN Microfiber Cloth - 4 pcs - 40x40 cms ...  car & motorbike   
  Involve Your Senses One Musk Organic Car Perfu...  car & motorbike   
  ShineXPro Microfiber Car Cleaning Cloth - OG S...  car & motorbike   
  pTron Bullet Pro 36W PD Quick Charger, 3 Port ...  car & motorbike   
  Pidilite WD-40, Multipurpose Car care Spray, 4...  car & motorbike   

                  sub_category  \
  All Car & Motorbike Products   
  All Car & Motorbike Products   
  All Car & Motorbike Products   
  All Car & Motorbike Products   
  All Car & Motorbike Products   

                                              image  \
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   

                                               link ratings no_of_ratings  \
  https://www.amazon.in/SOFTSPUN-Microfiber-Clea...     4.3        78,970   
  https://www.amazon.in/Involve-Your-Senses-IONE...     3.8        17,245   
  https://www.amazon.in/ShineXPro-Microfiber-Car...     4.3         5,067   
  https://www.amazon.in/PTron-Bullet-Pro-Lightwe...     4.0        15,910   
  https://www.amazon.in/Pidilite-Multiple-Mainte...     4.3        34,750   

 discount_price actual_price  Unnamed: 0  
           ₹269         ₹604         NaN  
           ₹307         ₹399         NaN  
           ₹499         ₹549         NaN  
           ₹349       ₹1,299         NaN  
           ₹290         ₹418         NaN

#B. Fitur Transformation

## 1.Hapus Kolom Unnamed : 0

In [142]:
concatedData1.drop('Unnamed: 0',axis=1,inplace=True) #menghapus kolom unnamed: 0 karena tdk diperlukan
concatedData1.head()

name    main_category  \
  SOFTSPUN Microfiber Cloth - 4 pcs - 40x40 cms ...  car & motorbike   
  Involve Your Senses One Musk Organic Car Perfu...  car & motorbike   
  ShineXPro Microfiber Car Cleaning Cloth - OG S...  car & motorbike   
  pTron Bullet Pro 36W PD Quick Charger, 3 Port ...  car & motorbike   
  Pidilite WD-40, Multipurpose Car care Spray, 4...  car & motorbike   

                  sub_category  \
  All Car & Motorbike Products   
  All Car & Motorbike Products   
  All Car & Motorbike Products   
  All Car & Motorbike Products   
  All Car & Motorbike Products   

                                              image  \
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   

                                               link ratings no_of_ratings  \
  https://www.amazon.in/SOFTSPUN-Microfiber-Clea...     4.3        78,970   
  https://www.amazon.in/Involve-Your-Senses-IONE...     3.8        17,245   
  https://www.amazon.in/ShineXPro-Microfiber-Car...     4.3         5,067   
  https://www.amazon.in/PTron-Bullet-Pro-Lightwe...     4.0        15,910   
  https://www.amazon.in/Pidilite-Multiple-Mainte...     4.3        34,750   

 discount_price actual_price  
           ₹269         ₹604  
           ₹307         ₹399  
           ₹499         ₹549  
           ₹349       ₹1,299  
           ₹290         ₹418

##2.Find and handle null Values

In [143]:
concatedData1.isna().sum() #cek apakah ada nilai null.

name                   0
main_category          0
sub_category           0
image                  0
link                   0
ratings           351588
no_of_ratings     351588
discount_price    122326
actual_price       35626
dtype: int64

dari hasil di atas, dapat diketahui yg terdapat null values adalah kolom ratings, no_of_ratings dan kolom discount_price, actual_price. untuk yg terkait ratings bisa dibiarkan saja, untuk yg terkait price, perlu diisi discount_price disamakan dg actual_price. Karena saya anggap tidak ada diskon.

In [144]:
concatedData1['discount_price'] = concatedData1['discount_price'].fillna(concatedData1['actual_price'])
concatedData1.isna().sum()

name                   0
main_category          0
sub_category           0
image                  0
link                   0
ratings           351588
no_of_ratings     351588
discount_price     35626
actual_price       35626
dtype: int64

setelah data discount_price yg null diisi dg value dari actual_price, dapat dilihat di atas bahwa jumlah row discunt_price yg null menurun drastis, dan sama dg actual_price

##3.Cek Datatype dan Ubah Ke yang lebih sesuai

###i. Ubah Rating ke float datatype

In [145]:
#karena ada error saat ubah data langsung ke float, mengindikasikan ada data selain numeric
#maka saya ubah dulu ke numeric datanya, dan yang error diubah jadi NaN (null) values sbb:
concatedData1['ratings'] = pd.to_numeric(concatedData1['ratings'], errors='coerce')

#kemudian saya ubah menjadi tipe data float
concatedData1['ratings'] = concatedData1['ratings'].astype(float)
concatedData1.info()


<class 'pandas.core.frame.DataFrame'>
Index: 1103170 entries,  to 
Data columns (total 9 columns):
 #   Column          Non-Null Count    Dtype  
---  ------          --------------    -----  
 0   name            1103170 non-null  object 
 1   main_category   1103170 non-null  object 
 2   sub_category    1103170 non-null  object 
 3   image           1103170 non-null  object 
 4   link            1103170 non-null  object 
 5   ratings         739116 non-null   float64
 6   no_of_ratings   751582 non-null   object 
 7   discount_price  1067544 non-null  object 
 8   actual_price    1067544 non-null  object 
dtypes: float64(1), object(8)
memory usage: 84.2+ MB


In [146]:
concatedData1.head()

name    main_category  \
  SOFTSPUN Microfiber Cloth - 4 pcs - 40x40 cms ...  car & motorbike   
  Involve Your Senses One Musk Organic Car Perfu...  car & motorbike   
  ShineXPro Microfiber Car Cleaning Cloth - OG S...  car & motorbike   
  pTron Bullet Pro 36W PD Quick Charger, 3 Port ...  car & motorbike   
  Pidilite WD-40, Multipurpose Car care Spray, 4...  car & motorbike   

                  sub_category  \
  All Car & Motorbike Products   
  All Car & Motorbike Products   
  All Car & Motorbike Products   
  All Car & Motorbike Products   
  All Car & Motorbike Products   

                                              image  \
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   

                                               link  ratings no_of_ratings  \
  https://www.amazon.in/SOFTSPUN-Microfiber-Clea...      4.3        78,970   
  https://www.amazon.in/Involve-Your-Senses-IONE...      3.8        17,245   
  https://www.amazon.in/ShineXPro-Microfiber-Car...      4.3         5,067   
  https://www.amazon.in/PTron-Bullet-Pro-Lightwe...      4.0        15,910   
  https://www.amazon.in/Pidilite-Multiple-Mainte...      4.3        34,750   

 discount_price actual_price  
           ₹269         ₹604  
           ₹307         ₹399  
           ₹499         ₹549  
           ₹349       ₹1,299  
           ₹290         ₹418

###ii. Ubah no_of_ratings ke float datatype

In [147]:
#saya hilangkan dulu nilai koma di data no_of_ratings
concatedData1['no_of_ratings'] = concatedData1['no_of_ratings'].str.replace(',', '')
concatedData1.head()

name    main_category  \
  SOFTSPUN Microfiber Cloth - 4 pcs - 40x40 cms ...  car & motorbike   
  Involve Your Senses One Musk Organic Car Perfu...  car & motorbike   
  ShineXPro Microfiber Car Cleaning Cloth - OG S...  car & motorbike   
  pTron Bullet Pro 36W PD Quick Charger, 3 Port ...  car & motorbike   
  Pidilite WD-40, Multipurpose Car care Spray, 4...  car & motorbike   

                  sub_category  \
  All Car & Motorbike Products   
  All Car & Motorbike Products   
  All Car & Motorbike Products   
  All Car & Motorbike Products   
  All Car & Motorbike Products   

                                              image  \
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   

                                               link  ratings no_of_ratings  \
  https://www.amazon.in/SOFTSPUN-Microfiber-Clea...      4.3         78970   
  https://www.amazon.in/Involve-Your-Senses-IONE...      3.8         17245   
  https://www.amazon.in/ShineXPro-Microfiber-Car...      4.3          5067   
  https://www.amazon.in/PTron-Bullet-Pro-Lightwe...      4.0         15910   
  https://www.amazon.in/Pidilite-Multiple-Mainte...      4.3         34750   

 discount_price actual_price  
           ₹269         ₹604  
           ₹307         ₹399  
           ₹499         ₹549  
           ₹349       ₹1,299  
           ₹290         ₹418

In [148]:
#karena ada error saat ubah data langsung ke float, mengindikasikan ada data selain numeric
#maka saya ubah dulu ke numeric datanya, dan yang error diubah jadi NaN (null) values sbb:
concatedData1['no_of_ratings'] = pd.to_numeric(concatedData1['no_of_ratings'], errors='coerce')

In [149]:
#kemudian saya ubah tipe data ke float
concatedData1['no_of_ratings'] = concatedData1['no_of_ratings'].astype(float)
concatedData1.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1103170 entries,  to 
Data columns (total 9 columns):
 #   Column          Non-Null Count    Dtype  
---  ------          --------------    -----  
 0   name            1103170 non-null  object 
 1   main_category   1103170 non-null  object 
 2   sub_category    1103170 non-null  object 
 3   image           1103170 non-null  object 
 4   link            1103170 non-null  object 
 5   ratings         739116 non-null   float64
 6   no_of_ratings   739053 non-null   float64
 7   discount_price  1067544 non-null  object 
 8   actual_price    1067544 non-null  object 
dtypes: float64(2), object(7)
memory usage: 84.2+ MB


###iii. Ubah discount_price ke float datatype

In [150]:
#cek apakah semua discount price dalam currency yg sama
concatedData1['discount_price'].str[0].unique()

array(['₹', nan], dtype=object)

dari hasil di atas, currency nya sama semua yaitu ₹, sehingga data bisa diolah atau dilakukan proses aggregate\

In [151]:
#saya hilangkan dulu nilai koma jika ada di data discount_price
concatedData1['discount_price'] = concatedData1['discount_price'].str.replace(',', '')
concatedData1.head()

name    main_category  \
  SOFTSPUN Microfiber Cloth - 4 pcs - 40x40 cms ...  car & motorbike   
  Involve Your Senses One Musk Organic Car Perfu...  car & motorbike   
  ShineXPro Microfiber Car Cleaning Cloth - OG S...  car & motorbike   
  pTron Bullet Pro 36W PD Quick Charger, 3 Port ...  car & motorbike   
  Pidilite WD-40, Multipurpose Car care Spray, 4...  car & motorbike   

                  sub_category  \
  All Car & Motorbike Products   
  All Car & Motorbike Products   
  All Car & Motorbike Products   
  All Car & Motorbike Products   
  All Car & Motorbike Products   

                                              image  \
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   

                                               link  ratings  no_of_ratings  \
  https://www.amazon.in/SOFTSPUN-Microfiber-Clea...      4.3        78970.0   
  https://www.amazon.in/Involve-Your-Senses-IONE...      3.8        17245.0   
  https://www.amazon.in/ShineXPro-Microfiber-Car...      4.3         5067.0   
  https://www.amazon.in/PTron-Bullet-Pro-Lightwe...      4.0        15910.0   
  https://www.amazon.in/Pidilite-Multiple-Mainte...      4.3        34750.0   

 discount_price actual_price  
           ₹269         ₹604  
           ₹307         ₹399  
           ₹499         ₹549  
           ₹349       ₹1,299  
           ₹290         ₹418

In [152]:
#extract nilai numeric saja di kolom discount_price
concatedData1['discount_price'] = concatedData1['discount_price'].str.extract(r'(\d+.\d+)')[0]

In [153]:
#ubah datatype
concatedData1['discount_price'] = concatedData1['discount_price'].astype(float)
concatedData1.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1103170 entries,  to 
Data columns (total 9 columns):
 #   Column          Non-Null Count    Dtype  
---  ------          --------------    -----  
 0   name            1103170 non-null  object 
 1   main_category   1103170 non-null  object 
 2   sub_category    1103170 non-null  object 
 3   image           1103170 non-null  object 
 4   link            1103170 non-null  object 
 5   ratings         739116 non-null   float64
 6   no_of_ratings   739053 non-null   float64
 7   discount_price  1061202 non-null  float64
 8   actual_price    1067544 non-null  object 
dtypes: float64(3), object(6)
memory usage: 84.2+ MB


###iv. Ubah actual_price ke float datatype

In [154]:
#cek apakah semua actual_price dalam currency yg sama
concatedData1['actual_price'].str[0].unique()

array(['₹', nan], dtype=object)

dari hasil di atas, currency nya sama semua yaitu ₹, sehingga data bisa diolah atau dilakukan proses aggregate

In [155]:
#saya hilangkan dulu nilai koma jika ada di data actual_price
concatedData1['actual_price'] = concatedData1['actual_price'].str.replace(',', '')
concatedData1.head()

name    main_category  \
  SOFTSPUN Microfiber Cloth - 4 pcs - 40x40 cms ...  car & motorbike   
  Involve Your Senses One Musk Organic Car Perfu...  car & motorbike   
  ShineXPro Microfiber Car Cleaning Cloth - OG S...  car & motorbike   
  pTron Bullet Pro 36W PD Quick Charger, 3 Port ...  car & motorbike   
  Pidilite WD-40, Multipurpose Car care Spray, 4...  car & motorbike   

                  sub_category  \
  All Car & Motorbike Products   
  All Car & Motorbike Products   
  All Car & Motorbike Products   
  All Car & Motorbike Products   
  All Car & Motorbike Products   

                                              image  \
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   

                                               link  ratings  no_of_ratings  \
  https://www.amazon.in/SOFTSPUN-Microfiber-Clea...      4.3        78970.0   
  https://www.amazon.in/Involve-Your-Senses-IONE...      3.8        17245.0   
  https://www.amazon.in/ShineXPro-Microfiber-Car...      4.3         5067.0   
  https://www.amazon.in/PTron-Bullet-Pro-Lightwe...      4.0        15910.0   
  https://www.amazon.in/Pidilite-Multiple-Mainte...      4.3        34750.0   

  discount_price actual_price  
           269.0         ₹604  
           307.0         ₹399  
           499.0         ₹549  
           349.0        ₹1299  
           290.0         ₹418

In [156]:
#extract nilai numeric saja di kolom actual_price
concatedData1['actual_price'] = concatedData1['actual_price'].str.extract(r'(\d+.\d+)')[0]

In [157]:
#ubah datatype actual_price ke float
concatedData1['actual_price'] = concatedData1['actual_price'].astype(float)
concatedData1.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1103170 entries,  to 
Data columns (total 9 columns):
 #   Column          Non-Null Count    Dtype  
---  ------          --------------    -----  
 0   name            1103170 non-null  object 
 1   main_category   1103170 non-null  object 
 2   sub_category    1103170 non-null  object 
 3   image           1103170 non-null  object 
 4   link            1103170 non-null  object 
 5   ratings         739116 non-null   float64
 6   no_of_ratings   739053 non-null   float64
 7   discount_price  1061202 non-null  float64
 8   actual_price    1065784 non-null  float64
dtypes: float64(4), object(5)
memory usage: 84.2+ MB


##4.Tambah Kolom discount_rate (hasil aggregate melibatkan discount_price dan actual_price)

In [158]:
concatedData1['discount_rate'] = concatedData1['discount_price'] / concatedData1['actual_price']
concatedData1.head()

name    main_category  \
  SOFTSPUN Microfiber Cloth - 4 pcs - 40x40 cms ...  car & motorbike   
  Involve Your Senses One Musk Organic Car Perfu...  car & motorbike   
  ShineXPro Microfiber Car Cleaning Cloth - OG S...  car & motorbike   
  pTron Bullet Pro 36W PD Quick Charger, 3 Port ...  car & motorbike   
  Pidilite WD-40, Multipurpose Car care Spray, 4...  car & motorbike   

                  sub_category  \
  All Car & Motorbike Products   
  All Car & Motorbike Products   
  All Car & Motorbike Products   
  All Car & Motorbike Products   
  All Car & Motorbike Products   

                                              image  \
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   
  https://m.media-amazon.com/images/W/IMAGERENDE...   

                                               link  ratings  no_of_ratings  \
  https://www.amazon.in/SOFTSPUN-Microfiber-Clea...      4.3        78970.0   
  https://www.amazon.in/Involve-Your-Senses-IONE...      3.8        17245.0   
  https://www.amazon.in/ShineXPro-Microfiber-Car...      4.3         5067.0   
  https://www.amazon.in/PTron-Bullet-Pro-Lightwe...      4.0        15910.0   
  https://www.amazon.in/Pidilite-Multiple-Mainte...      4.3        34750.0   

  discount_price  actual_price  discount_rate  
           269.0         604.0       0.445364  
           307.0         399.0       0.769424  
           499.0         549.0       0.908925  
           349.0        1299.0       0.268668  
           290.0         418.0       0.693780

#C.Data Demography

Overview :
1. Menggunakan describe, analisa statistik dari data numeric seperti mean, max, min nya untuk ratings, no_of_ratings, discount_price, actual_price
2. Analisa avg no_of_ratings berdasarkan sub category
3. Analisa avg no_of_ratings berdasarkan main category
4. Analisa ratings berdasarkan sub category
5. Analisa ratings berdasarkan main category
6. Analisa discount_rate berdasarkan sub category
7. Analisa discount_rate berdasarkan main category

## 1.Analisa statistik data numeric

In [159]:
concatedData1.describe()

ratings  no_of_ratings  discount_price  actual_price  \
count  739116.000000  739053.000000    1.061202e+06  1.065784e+06   
mean        3.832311     840.847039    2.889402e+03  2.314933e+04   
std         0.756101    8651.895695    9.591872e+03  1.356200e+07   
min         1.000000       1.000000    9.720000e+00  1.000000e+02   
25%         3.500000       4.000000    3.990000e+02  9.950000e+02   
50%         3.900000      20.000000    6.990000e+02  1.599000e+03   
75%         4.300000     133.000000    1.559000e+03  2.999000e+03   
max         5.000000  589547.000000    1.249990e+06  9.900000e+09   

       discount_rate  
count   1.061126e+06  
mean    5.478464e-01  
std     2.447630e-01  
min     5.949495e-08  
25%     3.561301e-01  
50%     4.998913e-01  
75%     7.141792e-01  
max     1.000000e+00

Dari hasil describe statistik data di atas, dapat dilihat bahwa
1. Total data produk yang ada adalah 1103170
2. Total data produk yg memiliki nilai ratings (tidak null) adalah 739116
3. Rata2 nilai ratings yang diberikan untuk setiap produk adalah 3,8 dari 5
4. Rata2 no_of_ratings (jumlah data rating yg diberikan) untuk setiap produk adalah 8652
5. Rata2 discount rate adalah 0,5478 , atau 54,78%

## 2.Analisa avg ratings berdasarkan sub category

In [160]:
concatedData1.groupby('sub_category')['ratings'].mean().sort_values(ascending=False)

sub_category
International Toy Store        4.578947
Value Bazaar                   4.304545
All Grocery & Gourmet Foods    4.176296
Health & Personal Care         4.174317
Strollers & Prams              4.127092
                                 ...   
Headphones                     3.503177
Car Electronics                3.497718
Personal Care Appliances       3.455056
Security Cameras               3.319143
Refurbished & Open Box         3.135000
Name: ratings, Length: 112, dtype: float64

Lima besar Rata-rata Rating Tertinggi masing2 didapat oleh product dg sub category
1. International Toy Store (4,58)
2. Value Bazaar	(4.304545)
3. All Grocery & Gourmet Foods	(4.176296)
4. Health & Personal Care	(4.174317)
5. Strollers & Prams	(4.127092)

## 3.Analisa avg ratings berdasarkan main_category

In [161]:
concatedData1.groupby('main_category')['ratings'].mean().sort_values(ascending=False)

main_category
grocery & gourmet foods    4.072739
toys & baby products       4.030417
pet supplies               4.029325
beauty & health            4.002085
home & kitchen             3.977118
music                      3.964741
stores                     3.946108
bags & luggage             3.939759
accessories                3.904224
kids' fashion              3.861064
appliances                 3.840975
industrial supplies        3.840392
tv, audio & cameras        3.825332
sports & fitness           3.805097
women's clothing           3.766114
men's clothing             3.765749
car & motorbike            3.727236
women's shoes              3.678979
men's shoes                3.638153
home, kitchen, pets        3.135000
Name: ratings, dtype: float64

dari hasil di atas, Lima besar rata-rata rating tertinggi didapat oleh main kategory sbb
1. grocery & gourmet foods	4.072739
2. toys & baby products	4.030417
3. pet supplies	4.029325
4. beauty & health	4.002085
5. home & kitchen	3.977118

## 4.Analisa avg no_of_ratings berdasarkan sub_category

In [162]:
concatedData1.groupby('sub_category')['no_of_ratings'].mean().sort_values(ascending=False)

sub_category
Value Bazaar                31114.606061
All Electronics              8881.106107
Baby Products                6371.507692
Toys Gifting Store           5352.375000
Headphones                   5040.975192
                                ...     
Jewellery                      59.963037
Fashion Sales & Deals          55.279070
Fashion Sandals                49.395153
Gold & Diamond Jewellery       24.241672
Refurbished & Open Box          4.200000
Name: no_of_ratings, Length: 112, dtype: float64

dari data di atas, Lima besar sub kategory produk yang mendapatkan rata-rata ulasan terbanyak (avg no_of_ratings terbanyak) adalah :
1. Value Bazaar	31114.606061
2. All Electronics	8881.106107
3. Baby Products	6371.507692
4. Toys Gifting Store	5352.375000
5. Headphones	5040.975192

## 5.Analisa avg no_of_ratings berdasarkan main_category

In [163]:
concatedData1.groupby('main_category')['no_of_ratings'].mean().sort_values(ascending=False)

main_category
tv, audio & cameras        3803.723694
toys & baby products       2628.869698
beauty & health            1354.834520
stores                      739.890006
home & kitchen              711.837689
pet supplies                680.694796
appliances                  660.270654
grocery & gourmet foods     544.663133
music                       401.163915
bags & luggage              386.524747
industrial supplies         374.385719
sports & fitness            321.656774
kids' fashion               273.845644
car & motorbike             268.582570
men's shoes                 223.531823
women's clothing            217.486930
women's shoes               195.822183
accessories                 172.371498
men's clothing              130.162498
home, kitchen, pets           4.200000
Name: no_of_ratings, dtype: float64

Lima besar kategori produk yg rata-rata popularitasnya tertinggi (paling banyak diulas) adalah sbb
1. tv, audio & cameras	3803.723694
2. toys & baby products	2628.869698
3. beauty & health	1354.834520
4. stores	739.890006
5. home & kitchen	711.837689

## 6.Analisa avg dicount_rate Berdasarkan Sub Category

In [164]:
concatedData1.groupby('sub_category')['discount_rate'].mean().sort_values(ascending=False)

sub_category
International Toy Store       0.878969
Sportswear                    0.851152
Coffee, Tea & Beverages       0.851132
Refurbished & Open Box        0.827663
Luxury Beauty                 0.816881
                                ...   
Western Wear                  0.441210
Jewellery                     0.425029
Clothing                      0.404683
Ethnic Wear                   0.372631
Fashion & Silver Jewellery    0.357572
Name: discount_rate, Length: 112, dtype: float64

Dari hasil di atas, Lima besar produk sub_category yang memiliki rata2 discount_rate tertinggi adalah sbb
1. International Toy Store	0.878969
2. Sportswear	0.851152
3. Coffee, Tea & Beverages	0.851132
4. Refurbished & Open Box	0.827663
5. Luxury Beauty	0.816881

## 7.Analisa avg discount_rate berdasarkan main_category

In [165]:
concatedData1.groupby('main_category')['discount_rate'].mean().sort_values(ascending=False)

main_category
home, kitchen, pets        0.827663
grocery & gourmet foods    0.810828
pet supplies               0.685823
beauty & health            0.685482
toys & baby products       0.649123
men's shoes                0.634033
music                      0.628229
industrial supplies        0.619343
appliances                 0.619165
women's shoes              0.606698
stores                     0.594841
men's clothing             0.587080
kids' fashion              0.578385
car & motorbike            0.567721
sports & fitness           0.555286
bags & luggage             0.543996
home & kitchen             0.542618
tv, audio & cameras        0.530690
accessories                0.495118
women's clothing           0.435311
Name: discount_rate, dtype: float64

Dari hasil di atas, dapat diketahui bahwa rata-rata discount tertinggi diberikan oleh produk-produk yang berada di main_Category sbb
1. home, kitchen, pets	0.827663
2. grocery & gourmet foods	0.810828
3. pet supplies	0.685823
4. beauty & health	0.685482
5. toys & baby products	0.649123